numpy/core/fromnumeric.py

```diff
@@ -1875,25 +1890,7 @@ def sum(a, axis=None, dtype=None, out=None, keepdims=np._NoValue):
     -128
 
     """
-    kwargs = {}
-    if keepdims is not np._NoValue:
-        kwargs['keepdims'] = keepdims
-    if isinstance(a, _gentype):
-        res = _sum_(a)
-        if out is not None:
-            out[...] = res
-            return out
-        return res
```

Comment:
You should leave this code here in `sum` and add:
```python
# Numpy 1.15, 2018-02-25
warnings.warn(
    "Calling np.sum(generator) is deprecated, and in the future will give a different result. "
    "Use np.sum(np.from_iter(generator)) or the python sum builtin instead."
    FutureWarning, stacklevel=2)
```

Reply:
Since it's undocumented, it should be okay to remove it entirely, no? I doubt much people rely on it.

Reply:
Unfortunately that's rarely the case. In this case, it's very likely that someone's old code is doing
```
from numpy import *

def use_what_I_thought_was_the_builtin_sum():
    return sum(i for i in range(10))
```
We can't just break their code without warning, much though it's tempting to.

Reply:
You're right. Done!

Reply:
Should we document the deprecation? I shouldn't think it's necessary, since the functionality itself wasn't documented in the first place.